video_xblock/video_xblock.py

```diff
@@ -0,0 +1,205 @@
+"""
+Video XBlock provides a convenient way to embed videos hosted on
+supported platforms into your course.
+All you need to provide is video url, this XBlock does the rest for you.
+"""
+
+import logging
+import pkg_resources
+
+from xblock.core import XBlock
+from xblock.fields import Scope, Boolean, Integer, Float, String
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from django.template import Template, Context
+
+from backends.base import BaseVideoPlayer, html_parser
+
+
+_ = lambda text: text
+log = logging.getLogger(__name__)
+
+
+class VideoXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    Main VideoXBlock class.
+    Responsible for saving video settings and rendering it for students.
+    """
+
+    icon_class = "video"
+
+    display_name = String(
+        default='Video',
+        display_name=_('Component Display Name'),
+        help=_('The name students see. This name appears in the course ribbon and as a header for the video.'),
+        scope=Scope.content,
+    )
+
+    href = String(
+        default='',
+        display_name=_('Video URL'),
+        help=_('URL of the video page. E.g. https://example.wistia.com/medias/12345abcde'),
+        scope=Scope.content
+    )
+
+    account_id = String(
+        default='',
+        display_name=_('Account Id'),
+        help=_('Your Brightcove account id'),
+        scope=Scope.content,
+    )
+
+    player_name = String(
+        default='dummy-player',
+        scope=Scope.content
+    )
+
+    # Playback state fields
+    current_time = Integer(  # Seconds played back from the start
+        default=0,
+        scope=Scope.user_state
+    )
+
+    playback_rate = Float(  # Video playback speed: 0.5, 1, 1.5, 2
+        default=1,
+        scope=Scope.preferences
+    )
+
+    volume = Float(  # Video volume: from 0 to 1
+        default=1,
+        scope=Scope.preferences
+    )
+
+    muted = Boolean(  # Video muted or not
+        default=False,
+        scope=Scope.preferences
+    )
+
+    editable_fields = ('display_name', 'href', 'account_id')
+    player_state_fields = ('current_time', 'muted', 'playback_rate', 'volume')
+
+    @property
+    def player_state(self):
```

Comment:
I think I understand it in general, why not use just function, not setter? We can talk about this.